Extension/src/githubAPI.ts

```diff
@@ -248,15 +248,18 @@ async function getRateLimit(): Promise<RateLimit> {
 }
 
 async function rateLimitExceeded(): Promise<boolean> {
-    const rateLimit: RateLimit = await getRateLimit();
-    return rateLimit && rateLimit.rate.remaining <= 0;
```

Comment:
I don't see why this line needs to change at all.  Is there a bug with the boolean expression?

Reply:
Yes.  After adding `| undefined` to the type, it does not like performing a check for the null/undefined and a boolean expression in the same statement.  The following:

```
return rateLimit && rateLimit.rate.remaining <= 0;
```

Results in:

```
Type 'boolean | undefined' is not assignable to type 'boolean'.
  Type 'undefined' is not assignable to type 'boolean'.
```

And the following:

```
return rateLimit?.rate.remaining <= 0;
```

Results in:

```
Object is possibly 'undefined'.
```

As the result of the left hand side may be undefined, which cannot be compared to 0.